src/RichXMLParser.m

```diff
@@ -59,13 +59,16 @@ -(BOOL)parseRichXML:(NSData *)xmlData
                                                         error:&error];
         if (xmlDocument == nil && error != nil) {
 			if ([error.domain isEqualToString:NSXMLParserErrorDomain]) {
-				if (error.code == NSXMLParserGTRequiredError)
-					return NO;
-				if (error.code == NSXMLParserPrematureDocumentEndError)
+			    // handle here cases identified to cause
+			    // application crashes caused by
+			    // NSXMLDocument's -initWithData:options:error
+			    // when option NSXMLDocumentTidyXML is enabled
+				switch (error.code) {
+				case NSXMLParserGTRequiredError:
 					return NO;
+				}
 			}
-			// Do we even want to try this?
-			// We get assertion failures with the above 2 cases, so we'll probably get them with other cases too.
+			// recover some cases like text encoding errors, non standard tags...
             xmlDocument = [[NSXMLDocument alloc] initWithData:xmlData
```

Comment:
For handling if this code produces an error:
```objective-c
xmlDocument = [[NSXMLDocument alloc] initWithData:xmlData
                                                  options: NSXMLDocumentTidyXML
                                                  error:&error];
```
Also we should nullify `error` before we call that otherwise won't `error != nil` even if successful?